src/app/services/BackendService.ts

```diff
@@ -0,0 +1,288 @@
+import { Injectable } from '@angular/core';
+import { ConnectedUserService } from './ConnectedUserService';
+import { environment } from 'src/environments/environment';
+import { Part } from '../domain/Part';
+import { MinimalUser } from '../domain/MinimalUser';
+import { Player, PlayerOrNone } from '../jscaip/Player';
+import { PlayerNumberMap } from '../jscaip/PlayerMap';
+import { ConfigRoom } from '../domain/ConfigRoom';
+import { Debug } from '../utils/Debug';
+import { JSONValue, MGPFallible, MGPOptional, MGPValidation, Utils } from '@everyboard/lib';
+import { Localized } from '../utils/LocaleUtils';
+
+export class BackendFailure {
+    public static readonly GAME_DOES_NOT_EXIST: Localized = () => $localize`This game does not exist!`;
+}
+
+type HTTPMethod = 'POST' | 'GET' | 'PATCH' | 'HEAD' | 'DELETE';
+
+@Injectable({
+    providedIn: 'root',
+})
+@Debug.log
+export class BackendService {
+    public constructor(private readonly connectedUserService: ConnectedUserService) {
+    }
+
+    private async performRequest(method: HTTPMethod, endpoint: string): Promise<MGPFallible<Response>> {
+        const token: string = await this.connectedUserService.getIdToken();
+        const response: Response =
+            await fetch(environment.backendURL + '/' + endpoint, {
+                method,
+                headers: {
+                    'Authorization': 'Bearer ' + token,
+                },
+            });
+        if (this.isSuccessStatus(response.status)) {
+            return MGPFallible.success(response);
+        } else {
+            try {
+                const jsonResponse: JSONValue = await response.json();
+                if (jsonResponse != null) {
+                    // eslint-disable-next-line dot-notation
+                    return MGPFallible.failure(jsonResponse['reason'] as string);
```

Comment:
what if jsonResponse['reason'] happened not to have stuff ? what's logged ?

Reply:
maintenant on check que jsonResponse['reason'] est pas null-like !